src/main/scala/beam/sim/vehicles/UniformVehiclesAdjustment.scala

```diff
@@ -12,16 +12,19 @@ case class UniformVehiclesAdjustment(beamServices: BeamServices) extends Vehicle
   private val realDistribution: UniformRealDistribution = new UniformRealDistribution()
   realDistribution.reseedRandomGenerator(beamServices.beamConfig.matsim.modules.global.randomSeed)
 
-  private val vehicleTypesAndProbabilitiesByCategory = beamServices.vehicleTypes.values.groupBy(_.vehicleCategory).map {
-    catAndType =>
-      val probSum = catAndType._2.map(_.sampleProbabilityWithinCategory).sum
-      val cumulProbs = catAndType._2
-        .map(_.sampleProbabilityWithinCategory / probSum)
-        .scan(0.0)(_ + _)
-        .drop(1)
-        .toList :+ 1.0
-      (catAndType._1, catAndType._2.zip(cumulProbs).map(pair => (pair._1, pair._2)))
-  }
+  private val vehicleTypesAndProbabilitiesByCategory: Map[VehicleCategory, Array[(BeamVehicleType, Double)]] =
```

Comment:
@colinsheppard, it's just rewritten with local vals and removed the usage of `._1` and `._2` to access the tuple (instead use `case` and `unapply`)